doc/ssdmond/ssdmond-hld.md

```diff
@@ -0,0 +1,129 @@
+
+1. `ssdmond` would be started by the `pmon` docker container
+2. The daemon would gather the static info once init-ed, by leveraging the `ssdutil` utility and update the StateDB
+3. It would periodically parse the priority 0 attributes either by leveraging `ssdutil` or directly through Linux utilities and update the StateDB.
+
+This is detailed in the sequence diagram below:
+
+![image.png](images/SSDMOND_SequenceDiagram.png)
+
+
+NOTE: While it is previously established that we use the abstraction provided by the `ssdutil` class to offer vendors the opportunity to implement their own SSD parsing logic, the [primary intent](#1-overview) of this iteration of the design is to enable streaming telemetry of these attributes, i.e., update the StateDB with the parsed data. A design choice is therefore made to bypass the abstraction logic in favor of this goal in the interim, with the expressed understanding that said abstraction will follow in a [future version](#future-work) of this daemon.
+
+
+### **2.4 Data Collection Logic**
+
+The SONiC OS already contains logic to parse information about SSDs from several vendors by way of the `ssdutil` platform utility. We leverage this utility to gather the following information:
+
+- Priority 0: Temperature
+- Priority 1: All aforementioned attributes
+
+This section will therefore only go into detail about data collection of attributes mentioned in [section 2.1](#21-priority-0-attributes):
+
+#### **2.4.1 IO Reads/Writes**
+
+- grep `/proc/diskstats` for statistics about the SSD of interest
+- Read the 4th value for reads completed successfully and the 8th value for writes completed<sup>[2](#2-kernelorg-procdiskstats)</sup>
+
+#### **2.4.2 Reserved Blocks Count**
+
+- Examine the SMART data for attributes related to reserved blocks or over-provisioning.
+- The exact attribute name and number can vary depending on the SSD manufacturer and model.
+- Look for keywords like "Reserved Block Count," "Over Provisioning," or similar terms.
+
+    Here's an example of what the SMART data output might look like:
+
+    ```
+    ...
+    ID# ATTRIBUTE_NAME          FLAGS    VALUE WORST THRESH FAIL RAW_VALUE
+    ...
+    173 Wear_Leveling_Count     0x0032   100   100   000    Old_age   Always       -       123
+    192 Power-Off_Retract_Count 0x0032   100   100   000    Old_age   Always       -       456
+    ...
+
+    ```
+
+    In this example, the "Wear_Leveling_Count" attribute might be indicative of reserved blocks or over-provisioning. However, the specific attribute and its interpretation can vary, so we make the search term and ID configurable by our vendors while maintaining a default search term in the event that this value is left unconfigured.
+
+## **StateDB Schema**
+```
+; Defines information for the SSD in a device
+
+key                 = SSD_INFO                       ; This key is for information that does not change for the lifetime of the SSD
```

Comment:
@assrinivasan may we should have the key indicating which SSD /dev/sdX?

Reply:
Addressed in latest commit @prgeor